src/infrastructure/implementations/blockstack-crux-user-repository.ts

```diff
@@ -0,0 +1,57 @@
+import { IKeyManager } from "src/core/interfaces/key-manager";
+import { CruxUser, SubdomainRegistrationStatus } from "../../core/entities/crux-user";
+import { ICruxBlockstackInfrastructure } from "../../core/interfaces";
+import {ICruxUserRepository, ICruxUserRepositoryOptions} from "../../core/interfaces/crux-user-repository";
+import {CruxDomainId, CruxId, IdTranslator} from "../../packages/identity-utils";
+import { getLogger } from "../../packages/logger";
+import { StorageService } from "../../packages/storage";
+import { BlockstackService } from "../services/blockstack-service";
+const log = getLogger(__filename);
+
+export interface IBlockstackCruxUserRepositoryOptions extends ICruxUserRepositoryOptions {
+    blockstackInfrastructure: ICruxBlockstackInfrastructure;
+    bnsOverrides?: string[];
```

Comment:
If this is not being used/provided by any client, can be removed.

Reply:
bnsOverrides are being used by wallet client now